### ISSUE TITLE
Flooding map for drivers

### ISSUE THREAD

**Issue by fileunderjeff** [sketch-city/project-ideas#28]
Create an app that drivers can use during or after rain storms in order to avoid roads that are likely to flood. Users can report flooded roads. App should also alert drivers if there is a risk of flash flooding, or a risk of flooding where their car is parked.

**Comment by patrickedelen** [comment]
I was actually thinking of doing something almost exactly like this for the hackathon when I saw it under the example projects. I have one other team member but if anyone else would like to help that would be awesome!

**Comment by fileunderjeff** [comment]
Awesome! This was suggested by Kristi Bartlett on the google doc.

**Comment by patrickedelen** [comment]
Cool! I'm sure the recent weather inspired a lot of ideas related to flooding.

**Comment by sana142** [comment]
Hi there, I just moved to Houston from Chicago, Il. I'm a graphic designer but I really like the idea of working with programmers and others in collaboration to create apps that can improve life quality. In Houston, I especially feel the value of something that could help with flooding and if you think you could benefit from a design perspective, I'd be interested! I think it's an exciting idea :D

**Comment by patrickedelen** [comment]
Hi Sana! I'm mostly backend myself so a front end designer would be awesome! I was thinking of making a webapp based on Esri's mapping api (check it out here https://developers.arcgis.com/javascript/3/) but I could always use some help with the logo and feel of the app.

**Comment by randy7771026** [comment]
Esri gave a bunch of credits last year.  Do not know if we have any this year.  Can you use OSM?

**Comment by patrickedelen** [comment]
I'm open to using anything right now. I was just gonna use Esri as I know a little about how their routing API is laid out. Does OSM have any back-end route processing like Esri so we could evaluate distance from reported flooding and stuff?

**Comment by randy7771026** [comment]
I think I remember you from last year.  What language do you want to use?

http://wiki.openstreetmap.org/wiki/Routing

**Comment by randy7771026** [comment]
You might could look at http://dmwilson.info/ archives or go back to HFD Active Incidents on the portal, map to rainfall records then predict that the same would flood under similar rainfall.

**Comment by patrickedelen** [comment]
I'd like to use Node.js serverside (I'm pretty familiar with JavaScript) and just HTML5 geolocation API + Javascript for GPS navigation on the front end so we don't have to build out any platform-specific code.

It does look like OSM would work as long as we can import the data from our maps and user reported locations.

**Comment by patrickedelen** [comment]
Wow the search tool on http://dmwilson.info/ is incredibly useful! If it's possible to export the data from those searches might be able to have a fairly good idea of what areas to avoid during flooding.

**Comment by randy7771026** [comment]
You might want to reach out to @dmwilson on the slack.

**Comment by randy7771026** [comment]
or on his github

**Comment by randy7771026** [comment]
https://github.com/d-m-wilson/HFDActiveIncidentsWeb

**Comment by patrickedelen** [comment]
Awesome I'll message him thanks!

**Comment by patrickedelen** [comment]
I created a new repo here: https://github.com/patrickedelen/FloodWatch/issues so those interested can plan a bit more.

**Comment by randy7771026** [comment]
I guess I will see you Saturday then.

**Comment by patrickedelen** [comment]
Yup! It's gonna be great!

**Comment by fileunderjeff** [comment]
@patrickedelen @randy7771026 @sana142 
I spoke with Nick at ESRI, and he sent me the following info:

Here is a code for 1000 ArcGIS Online credits:
- Developers can sign up for a free account here which gives them access to ArcGIS Online: https://developers.arcgis.com/sign-up/
- Once signed up they can then put the code in at the redeem voucher page: https://developers.arcgis.com/redeem-voucher/
- ArcGIS Online 1,000 credits code: HOUHACK16   (case sensitive)
- Valid Dates: 5/13/16 5:00pm Central – 5/27/16 5:00pm Central (code will work b/w these dates)

Also, we are having a Meetup on May 26th.  If some of your developers are interested in Geo they are welcome to attend - [GeoDev Meetup Houston](http://www.meetup.com/DevMeetUpTexas/events/230671230/)

**Comment by patrickedelen** [comment]
Thanks for the code @fileunderjeff ! @randy7771026 I looked into OSM and it seems that importing KML files into OSM maps is somewhat difficult but if we have Esri credits it shouldn't be too hard to switch over.

**Comment by d-m-wilson** [comment]
I'd like to help with this effort if I can, but I'm wondering about the utility of creating something that will fall apart when the ESRI credits run out. Am I missing something?

**Comment by patrickedelen** [comment]
@d-m-wilson good point; I have been looking around for other ways to do it and it might be possible with the Google Maps API (http://www.about-dev.com/javascript/41-find-places-close-by-a-route-using-google-maps-api-v3-and-routeboxer), just requires a lot more work to setup a spacial DB and stuff. 

ESRI does costs credits, but it seems like it would be a bit easier for our purposes. Regardless, I'm always up for a challenge!

**Comment by krisd1024** [comment]
Hi, I am Kristi, the one who submitted this project idea. I would love to collaborate with you guys on it if you're willing. I don't know how to build this, but I know a little about front-end design, and I know some about flooding/watershed analysis since I studied civil engineering.

**Comment by bryantw111** [comment]
I would like to collaborate on this as well, can help get water level data out of web pages. visualizing it and front-end design advice would be great! 

Created this API for data from the Harris county flood sensor system:
test.castaby.com:8080/htxHackers

**Comment by patrickedelen** [comment]
Hi @krisd1024 ! I think we could definitely use you on the team for front end stuff. @bryantw111 I believe we have 2-3 people (including myself) on data gathering right now and I'd like to make sure everyone has enough to do, but I'd like to include you if I can make sure you won't be bored the whole time :).

**Comment by krisd1024** [comment]
@patrickedelen are you already here? where are you sitting?

**Comment by patrickedelen** [comment]
I'm not there yet; I was planning on arriving around 11am.

**Comment by krisd1024** [comment]
Ok sounds good @patrickedelen I'll see you when you get here

**Comment by patrickedelen** [comment]
Awesome cya then @krisd1024 !

**Comment by krisd1024** [comment]
we are starting to work on this project in room 214 upstairs if others want to join when you arrive!

**Comment by krisd1024** [comment]
http://fas3.flood-alert.org/#Home:Home
Rice university and texas medical center flood alert system

**Comment by betterways** [comment]
also include biking trails & parks.  I know someone in the Bear Creek area that told me more than a week after the floods she hadn't biked to work because she hadn't had time to find out if her route was still under water or not.

**Comment by RawaaAl-Saffar** [comment]
Anyone interested on working on this project tomorrow?
